hr_contract_amendment/models/hr_contract.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2018 EBII (https://www.saaslys.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, _
+
+
+class HrAmendmentContractChanges(models.Model):
+    _name = 'hr.contract.amendment.changes'
+    _description = 'HR contract amendment change'
+
+    amendment = fields.Many2one(
+        comodel_name='hr.contract.amendment', string="Amendment",
+        Require=True, ondelete="cascade")
+    value_name = fields.Char(string="Contract attribute name")
+    value_before = fields.Char(string="Value before change")
+    value_new = fields.Char(string="New value")
+
+
+class HrAmendmentContract(models.Model):
+    _name = 'hr.contract.amendment'
+    _description = 'HR contract amendment'
+
+    contract_id = fields.Many2one(
+        comodel_name='hr.contract', string="Referenced Contract",
+        required=True)
+    amendment_number = fields.Char(string="Amendment name")
+    date = fields.Date(required=True, default=fields.Date.context_today)
+    user_id = fields.Many2one('res.users', string='Registred by',
+                              default=lambda self: self.env.user)
+    amendment_type = fields.Selection(selection=[
```

Comment:
for better inheritance, use a method to get the values

Reply:
seen many time in odoo code with 
    grep -rn --include="*.py" "Many2one('res.users'"
but have not found somethings like Many2one('res.users', compute="_get_current_user"
does an example or not understand what u mean